tests/integration-tests/framework/credential_providers.py

```diff
@@ -25,15 +25,20 @@ def register_cli_credentials_for_region(region, iam_role):
 
 def run_pcluster_command(*args, **kwargs):
     """Run a command after assuming the role configured through register_cli_credentials_for_region."""
+
     region = kwargs.get("region")
     if not region:
         region = os.environ["AWS_DEFAULT_REGION"]
 
-    if region in cli_credentials:
-        with sts_credential_provider(region, cli_credentials[region]):
+    if kwargs.get("custom_cli_credentials"):
+        with sts_credential_provider(region, credential_arn=kwargs.get("custom_cli_credentials")):
             return run_command(*args, **kwargs)
     else:
-        return run_command(*args, **kwargs)
+        if region in cli_credentials:
+            with sts_credential_provider(region, cli_credentials[region]):
+                return run_command(*args, **kwargs)
+        else:
+            return run_command(*args, **kwargs)
```

Comment:
I think this switching would be better done on the credentials argument itself and avoid multiple return statements.  It should be possible to do this with one assignment
```
credentials_arg = kwargs.get("custom_cli_credentials") or cli_credentials.get(region)
```
the `or` should do the switching for you, i.e. if kwargs.get("custom_cli_credentials") returns None, `credentials_arg` will be the region credentials

I think we should also refactor the cli_credentials[region] to use the get() method to avoid missing key errors.

Reply:
Will make the changes.